### DISCUSSION TITLE
nslookup DNS hijack

### DISCUSSION THREAD

**Discussion by dem1tris** [bol-van/zapret#292] [Q&A]
Всю голову сломал. Keenetic, entware.
Клиенты роутера нормально резолвят IP об любой DNS. Курл на DoH на кинетике тоже. Но nslookup выдает IP-заглушку для некоторых хостнеймов, и blockcheck на это тоже ругается.

Можете подсказать, почему так происходит?
```
~ # nslookup twitter.com
Server:    127.0.0.1
Address 1: 127.0.0.1 localhost

Name:      twitter.com
Address 1: 84.hi.jacked.43 host43.blah.blah.ru
```
```
~ # nslookup twitter.com 1.1.1.1
Server:    1.1.1.1
Address 1: 1.1.1.1 one.one.one.one

Name:      twitter.com
Address 1: 84.hi.jacked.43 host43.blah.blah.ru
```
```
~ # curl -H 'accept: application/dns-json' 'https://cloudflare-dns.com/dns-query?name=twitter.com&type=A'   
{"Status":0,"TC":false,"RD":true,"RA":true,"AD":false,"CD":false,"Question":[{"name":"twitter.com","type":1}],
"Answer":[{"name":"twitter.com","type":1,"TTL":417,"data":"104.244.42.129"}]}~
```

**Comment by bol-van** [comment]
Если заменить 127.0.0.1 на адрес роутера в локалке, тоже самое ?
Если с другого хоста сделать nslookup twitter.com <адрес_роутера_в_локалке>, что выдается ?

**Reply by dem1tris** [reply]
Поведение такое же:
из роутера по своему адресу в локалке -- IP заглушки
из клиента по адресу роутера в локалке -- честный адрес

**Reply by dem1tris** [reply]
Вопрос в целом уже чисто образовательный, т.к. стало ясно, что у провайдера всё самое нужное по IP заблочено)

**Reply by dem1tris** [reply]
Это видимо кэш какой-то. Выяснил, что роутерный nslookup на любой рандомный IP в качестве DNS-сервера отвечает одно и то же
